.github/workflows/ci.yml

```diff
@@ -0,0 +1,71 @@
+on:
+  pull_request:
+  workflow_dispatch:
+  push:
+    branches: [main, staging, trying]
+    tags:
+      - "v*.*.*"
+
+name: CI
+
+env:
+  CARGO_NET_RETRY: 3
+  CARGO_HTTP_CHECK_REVOKE: false
+
+jobs:
+  shellcheck:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Run ShellCheck
+        uses: azohra/shell-linter@v0.3.0
+
+  build:
+    name: Build Check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-rust
+
+      - name: Get Changed Files
+        id: files
+        uses: Ana06/get-changed-files@v2.1.0
+        with:
+          # use JSON so we don't have to worry about filenames with spaces
+          format: 'json'
+          filter: 'docker/Dockerfile.*'
+
+      - name: Validate Changed Images
```

Comment:
This might need a 120 minute timeout, since we use crosstool-ng images for most of these, which have quite a long build time.